src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintDashboardPanel.kt

```diff
@@ -105,6 +132,37 @@ class SonarLintDashboardPanel(private val editor: Editor) {
                 findingsSummaryLabel.text = fragments.joinToString()
             }
         }
+
+        val settings = Settings.getSettingsFor(project)
+        settings.connectionName?.let { connectionName ->
+            val serverConnection = getService(project, ProjectBindingManager::class.java).serverConnection
+
+            connectionLabel.text = "Connected to:"
+            connectionNameLabel.text = connectionName
+            connectionNameLabel.isVisible = true
+            connectionHelp.isVisible = false
+            connectionIcon.isVisible = true
+            if (serverConnection.isSonarCloud) {
+                connectionIcon.icon = SonarLintIcons.ICON_SONARCLOUD_16
+            } else {
+                connectionIcon.icon = SonarLintIcons.ICON_SONARQUBE_16
+            }
+            connectionIcon.icon = SonarLintIcons.ICON_SONARCLOUD_16
+            settings.projectKey?.let { projectKey ->
+                bindingLabel.isVisible = true
+                bindingLabel.text = "Bound to project: ${StringUtils.abbreviate(projectKey, 100)}"
+            } ?: run {
+                bindingLabel.isVisible = false
+                bindingLabel.text = NO_BINDING_TITLE
+            }
+        } ?: run {
+            connectionLabel.text = NO_CONNECTED_MODE_TITLE
+            connectionIcon.isVisible = false
+            connectionNameLabel.isVisible = false
+            connectionHelp.isVisible = true
+            bindingLabel.isVisible = false
+            bindingLabel.text = NO_BINDING_TITLE
+        }
```

Comment:
Bonus: shouldn't we also display the matched branch in connected mode?

Reply:
I was not able to make it work, fetching the branch information takes a lot of time and computation, it needs:
- Get selected file
- Get server connection
- Find module for the given file
- Get server branch name based on the module
It needs to be run on a pooled thread. I tried but I couldn't get the information displayed and I don't feel like we can spend more time on this for the scope of this sprint.